src/standalone/intellisense/resolveCompletionItem.ts

```diff
@@ -134,6 +145,36 @@ async function sendInspectRequest(
     kernel: Kernel.IKernelConnection,
     token: CancellationToken
 ): Promise<KernelMessage.IInspectReplyMsg['content']> {
+    const cacheKey = JSON.stringify(message);
+    const cache =
+        cachedKernelInspectRequests.get(kernel) || new Map<string, KernelMessage.IInspectReplyMsg['content']>();
+    cachedKernelInspectRequests.set(kernel, cache);
+    const cachedValue = cache.get(cacheKey);
+    if (cachedValue) {
+        return JSON.parse(JSON.stringify(cachedValue));
```

Comment:
Why is it nessecary to stringify and then parse instead of just use the cachedValue directly?

Reply:
To prevent this from getting altered, and as a result corrupting the cache.